File: scripts/analyze_pdf.py
import pdfplumber

pdf_path = r"C:\Users\Comunicaciones\Desktop\Moza\Proyecto Extreme V6.9\Reporte de Inventario-20260610-103507.pdf"

try:
    with pdfplumber.open(pdf_path) as pdf:
        for i, page in enumerate(pdf.pages):
            print(f"--- Page {i+1} ---")
            print(page.extract_text())
            print("\n")
except Exception as e:
    print(f"Error: {e}")

File: scripts/analyze_pdf.ts
import fs from 'fs';
import * as pdfjsLib from 'pdfjs-dist';

async function analyzePdf(filePath: string) {
  try {
    const data = new Uint8Array(fs.readFileSync(filePath));
    const loadingTask = pdfjsLib.getDocument({ data });
    const pdfDocument = await loadingTask.promise;
    
    let fullText = '';
    for (let i = 1; i <= pdfDocument.numPages; i++) {
      const page = await pdfDocument.getPage(i);
      const textContent = await page.getTextContent();
      const pageText = textContent.items.map((item: { str: string }) => item.str).join(' ');
      fullText += `--- Page ${i} ---\\n${pageText}\\n\\n`;
    }
    
    console.log('--- PDF TEXT START ---');
    console.log(fullText);
    console.log('--- PDF TEXT END ---');
  } catch (error) {
    console.error('Error reading PDF:', error);
  }
}

analyzePdf('C:\\Users\\Comunicaciones\\Desktop\\Moza\\Proyecto Extreme V6.9\\Reporte de Inventario-20260610-103507.pdf');
